app/controllers/topologies_controller.rb

```diff
@@ -59,6 +60,19 @@ def destroy
     redirect_to(topologies_url)
   end
 
+  # POST /topologies/:id/download_as_png
+  def download_as_png
+    filename = "#{ Rails.root }/tmp/#{ @topology.id }.png"
```

Comment:
I would prefer these go in a subdirectory, instead of being dumped into `tmp`. I suggest something like `tmp/topology_images`. Using [Tempfile](http://ruby-doc.org/stdlib-1.9.3/libdoc/tempfile/rdoc/Tempfile.html) is often cleaner, since you don't have to worry about where you save the image.

I think even better (haven't tested this myself, but it should work) is to have Rmagick generate the image and send it to the browser without having to save it to disk first:

```ruby
# ...
img[0].resize!(2)

send_data(
  img[0].to_blob(format: 'PNG'), 
  content_type: 'image/png', disposition: 'inline'
)
```